src/tests/ftest/util/apricot/apricot/test.py

```diff
@@ -132,17 +131,120 @@ def __init__(self, *args, **kwargs):
+            build_paths = json.load(build_vars)
+        self.basepath = os.path.normpath(os.path.join(build_paths['PREFIX'],
+                                                      '..') + os.path.sep)
+        self.prefix = build_paths['PREFIX']
+        try:
+            self.ofi_prefix = build_paths['OFI_PREFIX']
+        except KeyError:
+            self.ofi_prefix = os.sep + "usr"
+        self.cancel_from_list()
         self.check_variant_skip()
         self.log.info("*** SETUP running on %s ***", str(detect()))
         super(Test, self).setUp()
 
-    def check_variant_skip(self):
+    def add_test_data(self, filename, data):
+        """Add a file containing the specified data to the test variant
+           specific data directory.
+
+        Args:
+            filename (str): name of the file to create
+            data (list): data to write to the new file
+        """
+        try:
+            with open(os.path.join(self.outputdir, filename),
+                      "w") as file_handle:
+                file_handle.writelines(data)
+        except IOError as error:
+            self.fail("Error writing {}: {}".format(filename, error))
+
+    def cancel_from_list(self):
+        """Check if test is in skip list"""
+
+        def skip_process_error(message):
+            self.log.info(message)
+            self.log.info("Trudging on without skipping known failing tests")
+
+        def cancelForTicket(ticket, skip_list):
+            # put a copy of the skip-list into the logdir
+            self.add_test_data("skip-list", skip_list)
+            self.cancelForTicket(ticket)
+
+        try:
+            with open(os.path.join(os.sep, "scratch",
+                                   "CI-skip-list-master")) as skip_handle:
+                skip_list = skip_handle.readlines()
+        except Exception as excpt: # pylint: disable=broad-except
+            skip_process_error("Unable to read skip list: {}".format(excpt))
+            skip_list = []
+
+        for item in skip_list:
+            vals = item.split('|')
+            skip_it, ticket = self._check_variant_skip(literal_eval(vals[0]))
+            if skip_it:
+                # test is on the skiplist
+                if len(vals) > 1:
+                    # but there is a commit that fixes it
+                    try:
+                        with open(os.path.join(os.sep, 'tmp',
+                                               'commit_list')) as commit_handle:
+                            commits = commit_handle.readlines()
+                    except Exception as excpt: # pylint: disable=broad-except
+                        skip_process_error("Unable to read commit list: "
+                                           "{}".format(excpt))
+                        commits = None
+                    if commits and vals[1] in commits:
+                        # fix is in this code base
+                        self.log.info("This test variant is included in the "
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions